core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/ReorderJoins.java

```diff
@@ -603,6 +621,21 @@ static MultiJoinNode toMultiJoinNode(
                     Session session,
                     TypeAnalyzer typeAnalyzer,
                     TypeProvider types)
+            {
+                this(plannerContext, node, lookup, planNodeIdAllocator, sourceLimit, pushProjectionsThroughJoin, session, typeAnalyzer, types, false);
+            }
+
+            JoinNodeFlattener(
```

Comment:
Do we need this extra constructor in the Trino setup or its better to update all the references

Reply:
It'll be safe to keep it separate for now, since query fusion is the only call site for enabling `leftTreeOnly`